Code/Editor/EditorFramework/CodeGen/CppProject.cpp

```diff
@@ -321,54 +704,60 @@ ezResult ezCppProject::CompileSolution(const ezCppSettings& cfg)
   QApplication::setOverrideCursor(Qt::WaitCursor);
   EZ_SCOPE_EXIT(QApplication::restoreOverrideCursor());
 
-  EZ_LOG_BLOCK("Compile Solution");
-
-  if (cfg.m_sMsBuildPath.IsEmpty())
-  {
-    ezLog::Error("MSBuild path is not available.");
-    return EZ_FAILURE;
-  }
+  EZ_LOG_BLOCK("Compile C++ Plugin");
 
+  ezHybridArray<ezString, 32> errors;
+  ezInt32 iReturnCode = 0;
+#if EZ_ENABLED(EZ_PLATFORM_WINDOWS_DESKTOP)
   if (ezSystemInformation::IsDebuggerAttached())
   {
     ezQtUiServices::GetSingleton()->MessageBoxWarning("When a debugger is attached, MSBuild usually fails to compile the project.\n\nDetach the debugger now, then press OK to continue.");
   }
-
-  ezHybridArray<ezString, 32> errors;
+#endif
 
   ezProcessOptions po;
-  po.m_sProcess = cfg.m_sMsBuildPath;
+
+  ezString cmakePath = ezQtEditorApp::GetSingleton()->FindToolApplication(ezCppProject::GetCMakePath());
+
+  po.m_sProcess = cmakePath;
+  po.AddArgument("--build");
+  po.AddArgument(GetBuildDir(cfg));
+#if EZ_ENABLED(EZ_PLATFORM_WINDOWS)
+  const ezCppProject* preferences = ezPreferences::QueryPreferences<ezCppProject>();
+
+  if (preferences->m_CompilerPreferences.m_eCompiler == ezCompiler::Vs2022)
+  {
+    po.AddArgument("--config");
+    po.AddArgument(BUILDSYSTEM_BUILDTYPE);
+  }
+#endif
+  po.m_sWorkingDirectory = GetBuildDir(cfg);
   po.m_bHideConsoleWindow = true;
   po.m_onStdOut = [&](ezStringView sText) {
     if (sText.FindSubString_NoCase("error") != nullptr)
       errors.PushBack(sText);
   };
+  po.m_onStdError = [&](ezStringView sText) {
+    if (sText.FindSubString_NoCase("error") != nullptr)
+      errors.PushBack(sText);
+  };
 
-  po.AddArgument(ezCppProject::GetSolutionPath(cfg));
-  po.AddArgument("/m"); // multi-threaded compilation
-  po.AddArgument("/nr:false");
-  po.AddArgument("/t:Build");
-  po.AddArgument("/p:Configuration={}", BUILDSYSTEM_BUILDTYPE);
-  po.AddArgument("/p:Platform=x64");
-
-  ezStringBuilder sMsBuildCmd;
-  po.BuildCommandLineString(sMsBuildCmd);
-  ezLog::Dev("Running MSBuild: {}", sMsBuildCmd);
-
-  ezInt32 iReturnCode = 0;
+  ezStringBuilder sCMakeBuildCmd;
+  po.BuildCommandLineString(sCMakeBuildCmd);
+  ezLog::Dev("Running {} {}", cmakePath, sCMakeBuildCmd);
   if (ezProcess::Execute(po, &iReturnCode).Failed())
   {
-    ezLog::Error("MSBuild failed to run.");
+    ezLog::Error("Failed to start cmake.");
```

Comment:
In user visible texts we should write out the names properly, ie "CMake", "JSON", "HoloLens" etc
In code, I don't care that much.